octopus/src/main/scala/octopus/dsl.scala

```diff
@@ -58,118 +62,122 @@ object dsl {
     def ruleOption(pred: T => Option[Boolean], whenInvalid: String, whenNone: String): Validator[T] =
       compose(ValidationRules.ruleOption(pred, whenInvalid, whenNone))
 
+    def async(implicit appError: AppError[Future]): AsyncValidatorAsyncOps[Future, T] =
+      new AsyncValidatorAsyncOps[Future, T](AsyncValidatorM.lift[Future, T](v))
 
-    def async: AsyncValidatorAsyncOps[T] =
-      new AsyncValidatorAsyncOps[T](AsyncValidator.lift(v))
+    def asyncF[F[_]: AppError]: AsyncValidatorAsyncOps[F, T] =
```

Comment:
`asyncM` for consistency; let's also rename type parameter to `M` literally everywhere

Reply:
Done